MmSupervisorPkg/Core/Mem/SmmCpuMemoryManagement.c

```diff
@@ -563,14 +581,21 @@ ConvertMemoryPageAttributes (
     return RETURN_SUCCESS;
   }
 
-  PageTableBufferSize = 0;
-  WRITE_UNPROTECT_RO_PAGES (WriteProtect, CetEnabled);
-  Status = PageTableMap (&PageTableBase, PagingMode, NULL, &PageTableBufferSize, BaseAddress, Length, &PagingAttribute, &PagingAttrMask, IsModified);
+  while (!SamePageTable) {
+    PageTableBufferSize = 0;
+    WRITE_UNPROTECT_RO_PAGES (WriteProtect, CetEnabled);
+    Status = PageTableMap (&PageTableBase, PagingMode, NULL, &PageTableBufferSize, BaseAddress, Length, &PagingAttribute, &PagingAttrMask, IsModified);
+
+    if (Status == RETURN_BUFFER_TOO_SMALL) {
+      PageTableBuffer = AllocatePageTableMemory (EFI_SIZE_TO_PAGES (PageTableBufferSize), &SamePageTable);
+      ASSERT (PageTableBuffer != NULL);

```

Comment:
Yes, that keeps the same behavior as before.  Are you wondering if I should assert with SamePageTable as well or that we should fail instead of assert?

Reply:
To fail instead of assert. Looking at `PageTableMap()`, it looks like if this is null and `PageTableBufferSize` is non-zero, it will return `INVALID_PARAMETER`, exit this loop, and hit the condition with the print statement below. Assert again and exit.

If that's expected leave it as is. Otherwise, it seems like error handling could be added here to prevent calling `PageTableMap()` with an invalid (null) buffer pointer and make the error much more obvious with a message that says the page table could not be allocated due to a memory resources issue.

Reply:
I updated the logic to not do the second PageTableMap() call if the Buffer is NULL and also added a debug message for this case.  That'll keep the same failure behavior but better explain the error if we hit it.